Testreglar/4.1.2/App/app-4.1.2a.json

```diff
@@ -1,128 +1,237 @@
+			"ht": "<p>Slik tester du:</p>\n<ul>\n<li>Ta utgangspunkt i det tilgjengelige navnet du fant i forrige steg.</li>\n<li>Sjekk om det tilgjengelige navnet beskriver formålet med skjemaelementet. Det er tilstrekkelig at det tilgjengelige navnet identifiserer skjemaelementet.</li>\n</ul>",
+			"type": "jaNei",
+			"ruting": {
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.5"
+				},
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Nei",
+					"utfall": "Brukergrensesnittkomponenten har et tilgjengelig navn, som ikke beskriver formålet med komponenten."
+				}
+			}
+		},
+		{
+			"stegnr": "3.5",
+			"spm": "Er brukergrensesnittkomponenten kodet med riktig rolle?",
+			"ht": "<p>Slik tester du:</p>\n<ul>\n<li>Sjekk hvilken funksjonalitet komponenten har ved å trykke på den</li>\n<li>Sjekk om rollen som leses opp identifiserer den aktuelle komponentens funksjon</li>\n</ul>\n<p>Det betyr at knapp skal leses opp som knapp, og tilsvarende for andre komponenter.</p>\n<p><strong>Merk:</strong> Noen brukergrensesnittkomponenter ligger i en gruppe, som må leses i sammenheng med et overordnet spørsmål. Gruppen skal ha rolle \"group\" og du skal sjekke om skjermleser leser det opp i tillegg til rollen for den enkelte komponenten.</p>\n<p>Eksempel på elementer som ofte er knyttet til en gruppe: checkbox, combobox, listbox, menuitemcheckbox, menuitemradio, radio, searchbox, slider, spinbutton, switch, textbox.</p>",
+			"type": "jaNei",
+			"ruting": {
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Nei",
+					"utfall": "Brukergrensesnittkomponenten har et tilgjengelig navn som beskriver komponentens formål, men rollen identifiserer ikke komponentens funksjon."
+				},
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.6"
+				}
+			},
+			"kilde": [
+				"G10"
+			]
+		},
+		{
+			"stegnr": "3.6",
+			"spm": "Har brukergrensesnittkomponenten mer enn en tilstand når brukeren samhandler med den?",
+			"ht": "<p>En tilstand er en dynamisk egenskap som endres når brukeren samhandler med komponenten. Hvilke tilstander som er aktuelle vil variere med type komponent.</p>\n<p>Trykk eller dobbelttrykk (for å endre tilstand) på komponenten, og sjekk om den har minst én av disse tilstandene: </p>\n<ul>\n<li>av eller på</li>\n<li>spill av, pause, stopp</li>\n<li>aktivert eller deaktivert</li>\n<li>utvidet eller sammensluttet</li>\n<li>avkrysset eller ikke avkrysset</li>\n<li>markert eller ikke markert</li>\n<li>valgt eller ikke valgt</li>\n<li>nåværende</li>\n<li>gyldig eller ugyldig</li>\n<li>drag and drop</li>\n</ul>",
+			"type": "jaNei",
+			"ruting": {
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Ja",
+					"utfall": "Brukergrensesnittkomponenten har et tilgjengelig navn som beskriver komponentens formål, og rollen identifiserer komponentens funksjon."
+				},
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.7"
+				}
+			}
+		},
+		{
+			"stegnr": "3.7",
+			"spm": "Hvilken tilstand tester du?",
```

Comment:
Hvordan fungere dette viss skjemaelementet har mer enn en tilstand?

Reply:
Testregelen fungerer for flere tilstander. Hvis et element har kun en tilstand så avsluttes testen i steg 3.6.
Det skal registeres f.eks. av/på, valgt/ikke valgt osv.